spec/models/page_spec.rb

```diff
@@ -1333,6 +1333,18 @@ module Alchemy
           expect(contact.urlname).not_to match(/invisible/)
         end
 
+        context "with an invisible parent" do
+          before { parent.update_attribute(:visible, false) }
+
+          it "does not change if set_urlname is called" do
+            expect{page.send(:set_urlname)}.not_to change{page.urlname}
```

Comment:
Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.